screenshots/Screenshots.ts

```diff
@@ -1,48 +1,128 @@
+import childProcess from 'child_process';
+import fs from 'fs';
 import os from 'os';
+import path from 'path';
 
 import { expect } from '@playwright/test';
+import dayjs from 'dayjs';
 
 import { NavPage } from '../e2e/pages/nav-page';
 import { PreferencesPage } from '../e2e/pages/preferences';
 
-import type { Page, PageScreenshotOptions } from '@playwright/test';
+import type { Page } from '@playwright/test';
 
 interface ScreenshotsOptions {
   directory?: string;
+  isOsCommand?: boolean;
 }
 
 export class Screenshots {
+  private isOsCommand = true;
+  private sleepDuration = Number(process.env.RD_ENV_SCREENSHOT_SLEEP) || 1000;
+
+  // used by Mac api
+  private appBundleTitle = 'Electron';
+
+  protected windowTitle = '';
   private screenshotIndex = 0;
   readonly page: Page;
   readonly directory: string | undefined;
+  readonly screenshotDirectory: string | undefined;
 
   constructor(page: Page, opt?: ScreenshotsOptions) {
     this.page = page;
     if (opt?.directory) {
       const { directory } = opt;
 
       this.directory = directory === undefined || directory === null ? '' : `${ directory }/`;
+      this.screenshotDirectory = path.resolve(__dirname, 'output', os.platform(), this.directory);
+    }
+    if (opt?.isOsCommand) {
+      this.isOsCommand = opt.isOsCommand;
     }
   }
 
-  protected setPath(title: string, opt: PageScreenshotOptions = {}) {
-    return {
-      ...opt,
-      path: `screenshots/output/${ os.platform() }/${ this.directory }${ this.screenshotIndex++ }_${ title }.png`,
-    };
+  protected buildPath(title: string): string {
+    if (!this.screenshotDirectory) {
+      return '';
+    }
+
+    return path.resolve(this.screenshotDirectory, `${ this.screenshotIndex++ }_${ title }.png`);
+  }
+
+  protected async createScreenshotsDirectory() {
```

Comment:
Feel free to ignore this one if you want. We could just call the `mkdir` in `Screenshots.constructor` and not have to worry about calling it in subclasses. Your call though, I could be missing something here.

Reply:
I think I'll leave this one. I'm not a fan of constructors having side-effects. 

I'm also not a fan of the current implementation, but creating a directory if it doesn't exist before taking the screenshot seems more intuitive to me.